src/TriggerBinding/SqlTriggerTargetScaler.cs

```diff
@@ -14,24 +17,71 @@ internal sealed class SqlTriggerTargetScaler : ITargetScaler
         }
 
         public TargetScalerDescriptor TargetScalerDescriptor { get; }
 
         public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
         {
-            SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+            try
+            {
+                SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+
+                // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
+                int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
 
-            // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
-            int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
+                return GetScaleResultInternal(concurrency, metrics.UnprocessedChangeCount);
+            }
+            catch (Exception ex)
+            {
+                if (ex is SqlException sqlEx && (sqlEx.Message.Contains("Invalid object name") ||
+                                   (sqlEx.InnerException != null &&
+                                    sqlEx.InnerException.Message.Contains("Invalid object name"))))
+                {
+                    // If the SQL exception indicates that the object name is invalid,
+                    // we do not have the global state and leases table created.
+                    // Check if we already tried to spin up the worker that starts the listener which will create those tables.
+                    // If we have already tried, we will return 0 as the target worker count.
+                    this._logger.LogWarning("Invalid object name detected. SQL trigger tables not found.");
+                    if (_lastTableCheck != DateTime.MinValue)
+                    {
+                        // Check if there are any changes in the user table.
+                        int changes = await GetNumberOfChangesAsync(this._connectionString, this._userTable, this._logger, CancellationToken.None);
+                        // If there are changes in the user table, we need to spin up a worker to create the global state and leases table.
+                        if (changes > 0)
+                        {
+                            this._logger.LogWarning("There are changes in the change-tracking table for the user table, but the global state and leases table are not created. Spinning up a worker to create those tables.");
+                            _lastTableCheck = DateTime.UtcNow;
+                            return new TargetScalerResult
+                            {
+                                TargetWorkerCount = 1
```

Comment:
If the scale controller is checking every 10sec that means that we'll tell it to scale up once, but then 10sec later immediately tell it to scale down to 0 again. It's very unlikely that the tables will have been created by then, which could cause problems.

We discussed returning a target count of 1 for 1-2min to give more time for the function to be started and create the tables, so I think you'll want to modify the logic to account for that.

@alrod do you agree with that?

Reply:
I remember it being every minute and I thought that's time enough - but I could add threshold time to the check.